devices/oneplus-bacon/kernel/default.nix

```diff
@@ -0,0 +1,62 @@
+{
+  mobile-nixos
+, fetchFromGitHub
+, kernelPatches ? [] # FIXME
+, buildPackages
+, dtbTool
+}:
+
+(mobile-nixos.kernel-builder-gcc6 {
+  configfile = ./config.armv7l;
+
+  file = "zImage-dtb";
+  hasDTB = false;
+
+  version = "3.4.113";
+  src = fetchFromGitHub {
+    owner = "LineageOS";
+    repo = "android_kernel_oppo_msm8974";
+    rev = "a160ed58b3b94f63d1983b1798e280ed6602656a"; # lineageos-16.0
+    sha256 = "sha256-odpo3vnQ1M2eWzqEFrXtipQZxlL5ko0/tfDM5yUplMc=";
+  };
+  patches = [
+    #./99_framebuffer.patch
+    ./02_gpu-msm-fix-gcc5-compile.patch
+    ./0003-arch-arm-Add-config-option-to-fix-bootloader-cmdli.patch
+    ./99_framebuffer.patch
+  ];
+
+  isModular = false;
+
+}).overrideAttrs({ postInstall ? "", postPatch ? "", ... }: {
+  preConfigure = ''
+    makeFlagsArray+=("CONFIG_NO_ERROR_ON_MISMATCH=y")
+    buildFlagsArray+=("CONFIG_NO_ERROR_ON_MISMATCH=y")
```

Comment:
`CONFIG_` entries belong in the kernel `config.armv7l` file. (Unless they can't work there?)

Reply:
It doesn't work with this entry in the kernel config file.

Reply:
I see, [this is indeed a special case](https://github.com/LineageOS/android_kernel_oppo_msm8974/blob/a160ed58b3b94f63d1983b1798e280ed6602656a/scripts/Makefile.modpost#L83), this is the only location where the setting is used, there is no definition. Its use of `^CONFIG_` naming is quite the red herring!